src/Markup/Avalonia.Markup.Xaml/XamlIl/Runtime/RuntimePlatformInfo.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Runtime.InteropServices;
+
+
+namespace Avalonia.Markup.Xaml.XamlIl.Runtime;
+
+public class RuntimePlatformInfo
+{
+    private static OSPlatform IOS { get; } = OSPlatform.Create("IOS");
+    
+    private static OSPlatform Android { get; } = OSPlatform.Create("ANDROID");
+    
+    private static OSPlatform Browser { get; } = OSPlatform.Create("BROWSER");
+    
+    public static RuntimePlatformInfo Instance { get; } = new();
+    
+    private RuntimePlatformInfo()
+    {
+    }
+
+    public bool IsWindows => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+    public bool IsMacOS => RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+
+    public bool IsLinux => RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+
+    public bool IsIOS => RuntimeInformation.IsOSPlatform(IOS);
+
+    public bool IsAndroid => RuntimeInformation.IsOSPlatform(Android);
+
+    public bool IsBrowser => RuntimeInformation.IsOSPlatform(Browser);
+
+    public bool IsDesktop(Size primaryScreenSize) => primaryScreenSize.Width > 1280;
+
+    public bool IsLaptopOrDesktop(Size primaryScreenSize) =>
+        primaryScreenSize.Width > 1024 && primaryScreenSize.Width <= 1280;
+
+    public bool IsTablet(Size primaryScreenSize) =>
+        primaryScreenSize.Width > 768 && primaryScreenSize.Width <= 1024;
+
+    public bool IsMobileLandscape(Size primaryScreenSize) =>
+        primaryScreenSize.Width > 480 && primaryScreenSize.Width <= 768;
+
+    public bool IsMobile(Size primaryScreenSize) =>
+        primaryScreenSize.Width <= 480;
```

Comment:
Where is it used? We can never guarantee correct device type everywhere. Instead, we might have "Idiom" ported from MAUI/XF as well (they return correct device type when it's known from system API, and return "Unknown" otherwise, without relying on screen size).

Reply:
Idiom concept is broken by design. There is no such thing as a fixed idiom. An android phone can be connected to a 30" monitor with physical keyboard while your app is running. An app running on tablet can be switched to split-screen mode and effectively get screen space of a phone. Convertible 2-in-1 tablets exist. Etc, etc.

CSS doesn't introduce a "form-factor" concept for a reason.